python/ray/data/dataset.py

```diff
@@ -2986,6 +2986,7 @@ def iter_batches(
         local_shuffle_buffer_size: Optional[int] = None,
         local_shuffle_seed: Optional[int] = None,
         _collate_fn: Optional[Callable[[DataBatch], Any]] = None,
+        _finalize_fn: Optional[Callable[[DataBatch], Any]] = None,
```

Comment:
should we also make this addition at the top-level Dataset API as well? is there a separate formal process for such API changes?